vit_vqgan/data.py

```diff
@@ -100,11 +100,27 @@ def _normalize(image):
         ):
             if folder is not None:
                 # load files
-                files = [f"{Path(f)}" for f in Path(folder).glob("*.tfrecord")]
+                if "gs://" in folder:
+                    files = tfio.gfile.list_files(f"{folder}/*.tfrecord")
+                else:
+                    files = [f"{Path(f)}" for f in Path(folder).glob("*.tfrecord")]
                 assert len(files) > 0, f"No files found at folder: {folder}"
 
+                # shuffle files
+                if augment:
+                    random.shuffle(files)
+
                 # load dataset
-                ds = tf.data.TFRecordDataset(files)
+                ds = tf.data.TFRecordDataset(
+                    files,
+                    num_parallel_reads=tf.data.experimental.AUTOTUNE if augment else None,
+                )
+
+                # non deterministic read (faster)
+                if augment:
+                    ignore_order = tf.data.Options()
+                    ignore_order.experimental_deterministic = False
```

Comment:
nit: `experimental_deterministic` is deprecated according to the documentation, it should be enough to use `deterministic`.

Reply:
Yeah, flat files is much slower in term of disk access (same if you put on bucket).

Other alternative I liked was webdataset but it requires hacking around (starting process as `fork`) to work well on TPU's so I decided to use the recommended way which should be optimized on TPU's (even though I'm not a big fan of the format as you have to know the keys used on that file to decode it).

`Datapipes` seems really nice but I'm guessing it will have the same multiprocessing issue on TPU's.